### PR TITLE
TG not to accept events while re-provisioning

### LINKED ISSUE #390: Dataset removal not to kill Jena
It looks that dataset removal process still causes Jena to consume too many resources and so it's very slow.

**Acceptance criteria:**
- [x] the query checking if there are some triples to be deleted not to do a count but simply select a single non `CLIVersion` triple;
- [x] verify if the limit on the delete query works and maybe the solution from the point above could work better for the delete query, too;
- [x] add a semaphore to the `POST /events` (TG instances can be using some special triple in the Triples Store for synchronization) so:
  - [x] TG starting the re-provisioning process to set the special triple so it can be used by the other instances;
  - [x] TG finishing the re-provisioning process to clear the special triple once the process is finished;
  - [x] it responds with [`503 Service Unavailable`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) as long as the removal process is ongoing
  - [x] returns back to the operational state once the removal is finished;
  - [x] adapt event-log subscription mechanism so the `503`s cause putting a subscriber on hold;
  - [x] TG controlling re-provisioning to notify event-log about availability when the process is done;
  - [x] TG not controlling the re-provisioning process to notify event-log about availability when the flag is cleared;
  - [x] state of the re-provisioning flag to be cached;
- [x] use the CliVersion JSON object props in all the places;
- [x] remove the initial delay;
- [x] make a bugfix release.

### PR DESCRIPTION
**Pull request by jachro**
closes #390

### CODE REVIEW

File: triples-generator/src/main/scala/ch/datascience/triplesgenerator/reprovisioning/TriplesRemover.scala [hunk 1]
```diff
@@ -29,47 +32,43 @@ private trait TriplesRemover[Interpretation[_]] {
...
-
   override def removeAllTriples(): IO[Unit] =
-    queryExpecting[Long](findTriplesCount)(countDecoder) flatMap {
-      case 0 => IO.unit
-      case _ =>
+    queryExpecting(checkIfEmpty)(storeEmptyFlagDecoder) flatMap { isEmpty =>
+      if (isEmpty) IO.unit
+      else
         for {
           _ <- updateWitNoResult(removeTriplesBatch)
```
Comment (travissarles) [inline triples-generator/src/main/scala/ch/datascience/triplesgenerator/reprovisioning/TriplesRemover.scala:52]: missing an 'h' in updateWit**h**NoResult

File: acceptance-tests/src/test/resources/application.conf [hunk 2]
```diff
@@ -9,7 +9,7 @@ projects-tokens {
   db-url-template = "jdbc:h2:mem:$dbName;DB_CLOSE_DELAY=-1;MODE=PostgreSQL"
 }
 
-threads-number = 3
+threads-number = 4
```
[the remark this answers is not in the document]
Comment (jachro) [reply acceptance-tests/src/test/resources/application.conf:12]: This is purely for the acceptance-tests and it's a bit empirical value. I remember the tests weren't passing if the value was much higher (but please don't ask me what was that value).

File: event-log/src/test/scala/io/renku/eventlog/subscriptions/EventsSenderSpec.scala [hunk 3]
```diff
@@ -72,7 +76,7 @@ class EventsSenderSpec extends AnyWordSpec with ExternalServiceStubbing with sho
     }
 
     "return Misdelivered if call to the remote fails with Connect Exception" in new TestCase {
-      override val sender = new IOEventsSender(TestLogger())
+      override val sender = new IOEventsSender(TestLogger(), retryInterval = 10 millis)
```
[the remark this answers is not in the document]
Comment (jachro) [reply event-log/src/test/scala/io/renku/eventlog/subscriptions/EventsSenderSpec.scala:79]: Yes, 10 ms is just to make the tests verifying the logic but be as fast as possible.

File: triples-generator/src/main/scala/ch/datascience/triplesgenerator/reprovisioning/TriplesRemover.scala (hunk 1 above)
Comment (jachro) [reply triples-generator/src/main/scala/ch/datascience/triplesgenerator/reprovisioning/TriplesRemover.scala:52]: Done

File: triples-generator/src/test/scala/ch/datascience/triplesgenerator/reprovisioning/ReProvisioningStatusSpec.scala [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      findStatus shouldBe None
+    }
+
+    "not throw an error if the ReProvisioning object isn't there" in new TestCase {
+
+      findStatus shouldBe None
+
+      expectNotificationSent
+
+      reProvisioningStatus.clear().unsafeRunSync() should be((): Unit)
```
[the remark this answers is not in the document]
Comment (jachro) [reply triples-generator/src/test/scala/ch/datascience/triplesgenerator/reprovisioning/ReProvisioningStatusSpec.scala:74]: Indeed

**Review by travissarles — APPROVED** [review]

**Pull request merged**